docs/specs/kemsphinx.rst

```diff
@@ -30,7 +32,69 @@ has two functions:
 Therefore we must embed these KEM ciphertexts in the KEMSphinx header,
 one KEM ciphertext per mix hop.
 
-2. KEMSphinx Header Design
+2. Post Quantum Hybrid KEM
+==========================
+
+Special care must be taken in order correctly compose a hybrid post
+quantum KEM that is IND-CCA2 robust.
+
+The hybrid post quantum KEMs found in Cloudflare's circl library are
+suitable to be used with Noise or TLS but not with KEM Sphinx because
+they are not IND-CCA2 robust. Noise and TLS achieve IND-CCA2 security
+by mixing in the public keys and ciphertexts into the hash object and
+therefore do not require an IND-CCA2 KEM.
+
+Firstly, our post quantum KEM is IND-CCA2 however we must specifically take care
+to make our NIKE to KEM adapter have semantic security. Secondly, we must make a
+security preserving KEM combiner.
+
+2.1 NIKE to KEM adapter
+-----------------------
+
+We easily achieve our IND-CCA2 security by means of hashing
+together the DH shared secret along with both of the public keys:
+
+```
+func Encap(their_pubkey publickey) ([]byte, []byte) {
+        my_privkey, my_pubKey = GEN_KEYPAIR(RNG)
+        ss = DH(my_privkey, their_pubkey)
+        ss2 = H(ss || my_pubkey || their_pubkey)
+	return my_pubkey, ss2
+}
+
+func Decap(my_privkey, ciphertext) []byte {
+        s = DH(my_privkey, ciphertext)
```

Comment:
1. The `DH` used here is unclear. In `Encap` it' used like old-school DH with public+private keys, in `Decap` it takes `ciphertext` ? What is `ciphertext` here? Is this `KEM-DECRYPT` rather than DH or should `ciphertext` have read `their_pubkey` ?
2. `Encap` and `Decap` as outlined here produce different secrets:
    ```
    Encap_ss2 = H(ss || a_pubkey || b_pubkey)
    Decap_ss2 = H(ss || b_pubkey || ciphertext) # ciphertext=a_pubkey given 1)
    ```
    the order of the pubkeys should be flipped in one of them so it's consistent.

Reply:
here ciphertext is just another public key but i will rename it so that it's more clear

Reply:
fixed